src/views/Generator/RuleEditor/CorrelationEditor.tsx

```diff
@@ -25,6 +28,14 @@ export function CorrelationEditor() {
     }
   }
 
+  const extractedValue = useMemo(() => {
+    if (selectedRuleInstance?.type !== 'correlation') {
+      return
+    }
+
+    return selectedRuleInstance.state.extractedValue
+  }, [selectedRuleInstance])
```

Comment:
`selectedRuleInstance` is already `useMemo`'ed, so this can probably be removed in favor of
```ts
const extractedValue = selectedRuleInstance?.type !== 'correlation' ? selectedRuleInstance.state.extractedValue : undefined
```

Reply:
Good point, I've also added invariant to narrow down selected rule instance type